### PR TITLE
User cannot checkout before login

### PR DESCRIPTION
**Pull request by janderson16**
completed functionality & happy/sad path feature tests for:

user can create account and log in with valid credentials
user cannot log in without valid credentials or unique email
user cannot proceed to checkout before logging in
 -> checkout link needs to be updated when order functionality is built

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -14,8 +14,12 @@ def create
       flash[:notice] = "Welcome to PogHogs #{@user.first_name}"
       session[:user_id] = @user.id
       redirect_to '/dashboard'
+    elsif User.exists?(['email LIKE ?', "%#{@user.email}%"])
+      flash[:notice] = "Email already registered with an account!"
```
Comment (janderson16) [inline app/controllers/users_controller.rb:18]: informs user if their email account is on file

File: app/models/user.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 class User < ApplicationRecord
+  validates :email, presence: true, uniqueness: true
```
Comment (janderson16) [inline app/models/user.rb:2]: added validations for user info - validation testing covered in user feature tests

File: app/views/carts/show.html.erb [hunk 3]
```diff
@@ -4,3 +4,10 @@
 Item: <%= Item.find(id).title %><br>
 Quantity: <%= quantity %><br>
 <% end %>
+
+<%- unless current_user == nil -%>
+  <%= button_to "Proceed to checkout"%>
```
Comment (janderson16) [inline app/views/carts/show.html.erb:9]: !!!!! dead button !!!!! needs to be updated after checkout/functionality is implemented

**Review by janderson16 — COMMENTED** [review]
7 user_spec examples, all passing

**Comment by alfosco** [comment]
Well written tests, thanks for including sad paths.

**Pull request merged**